core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1129,6 +1138,8 @@ public void onRefreshSettings(Settings settings) {
             searchService.onRefreshSettings(settings);
             indexingService.onRefreshSettings(settings);
             if (change) {
+                engine().onSettingsChanged();
+                // TODO: why force a refresh here...?
```

Comment:
> doesn't it follow the same logic as in updateBufferSize ?

Hmm but `index_buffer_size` is not a live setting?

I think the `index.gc_deletes` and `index.version_map_size` changes should be separately handled ... I'll fix those explicitly, and remove the refresh.

I think it's sloppy to couple `refresh` to settings changes unless there really are settings changes that can only "become live" with a refresh.

Separately, `index.translog.fs.type` is a live setting, but we don't force a flush if that one is changed ... hmmm.